extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTableRefreshing.java

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.table.impl.PartitionAwareSourceTable;
+import io.deephaven.engine.table.impl.SourceTableComponentFactory;
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.engine.updategraph.UpdateSourceRegistrar;
+import io.deephaven.iceberg.layout.IcebergTableLocationProviderBase;
+import io.deephaven.iceberg.location.IcebergTableLocationKey;
+import org.apache.iceberg.Snapshot;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Table wrapper for refreshing Iceberg tables.
+ */
+public class IcebergTableRefreshing extends PartitionAwareSourceTable implements IcebergTable {
```

Comment:
If you're going to have the TLP have an interface for `update`, etc, we combine refreshing and static here, and delegate to the TLP for throwing an error or delivering an update.